src/model/newgens.jl

```diff
@@ -1,75 +1,88 @@
+function append_builds!(config, data, table_name, build_table_name)
+    if !haskey(data, build_table_name)
+        @warn "No table $build_table_name defined, not appending any builds to the $table_name table"
+        return
+    end
+    build = get_table(data, build_table_name)
     bus = get_table(data, :bus)
-    gen = get_table(data, :gen)
+    existing = get_table(data, table_name)
     years = get_years(data)
 
-    newgen_cols = Symbol.(names(newgen))
-
     # Filter any already-built exogenous generators
-    filter!(build_gen) do row
+    filter!(build) do row
         if row.build_type == "exog"
             row.year_on <= config[:year_gen_data] && return false
             row.year_on > last(years) && return false
-        end
-        
+        end 
         return true
     end
+    build.build_idx = 1:nrow(build)
+
+    # Set up a table to add to the gen table
+    new = make_new_from_build(config, data, build_table_name)
 
-    #get the names of specifications that will be pulled from the build_gen table
-    spec_names = filter!(
-        !in((:bus_idx, :gen_latitude, :gen_longitude, :gen_state, :gen_county, :reg_factor, :year_off, :year_shutdown, :pcap_inv, :year_unbuilt, :past_invest_cost, :past_invest_subsidy)), 
-        propertynames(newgen)
-    ) #this needs to be updated if there is anything else in gen that isn't a spec
+    # Begin filling in the other columns.
+    leftjoin!(new, build, on=:build_idx, makeunique=true)
 
-    for n in spec_names
-        hasproperty(build_gen, n) || error("Gen table has column $n, but not found in build_gen table. If this is a mod specific column, consider adding it to build_gen using modify_raw_data!()")
+    # Add custom columns:
+    new[!, "year_shutdown"]       .= "y9999" # add_to_year(year, spec_row.age_shutdown)
+    new[!, "year_off"]            .= "y9999"
+    new[!, "pcap_inv"]            .= 0.0
+    new[!, "past_invest_cost"]    .= Ref(Container(0.0))
+    new[!, "past_invest_subsidy"] .= Ref(Container(0.0))
+
+    new_bus_idxs = new.bus_idx::Vector{Int64}
+    
+    # Loop through and add all the other columns from the existing table
+    for col_name in names(existing)
+        # If the column already exists in `new`, impute from `bus` if it is a `bus` column and there are missing values
+        if hasproperty(new, col_name)
+            if hasproperty(bus, col_name)
+                col = new[!, col_name]
+                for (i, val) in enumerate(col)
+                    ismissing(val) || isnothing(val) || isempty(val) || continue
+                    bus_idx = new_bus_idxs[i]
+                    col[i] = bus[col_name, bus_idx]
+                end
+            end
+            continue
+        end
+
+        # If the column needs to come solely from the bus table, add it here.
+        if startswith(col_name, "bus_") || hasproperty(bus, col_name)
+            bus_col = hasproperty(bus, col_name) ? bus[!, col_name] : bus[!, "bus_$col_name"]
```

Comment:
my understanding was that in the "bus_" prefix would just be for columns in the gen table so there wouldn't be columns like `bus_$col_name` in the bus table, except for maybe bus_idx. Is this else case just for bus_idx? I would think the else case would be `col_name[5:end]` to look for a column in the bus table that is the same but without the bus_ prefix

Reply:
You are totally right, I will change and push that.  We aren't using this else case in the E4ST.jl tests but we will definitely hit it with our RFF inputs.